src/stan_math_backend/Transform_Mir.ml

```diff
@@ -3,18 +3,49 @@ open Middle
 
 let use_opencl = ref false
 
-let opencl_triggers =
+let opencl_trigger_restrictions =
   String.Map.of_alist_exn
-    [ ( "normal_id_glm_lpdf"
-      , [ (* Array of conditions under which to move to OpenCL *)
-          [(1, UnsizedType.UMatrix)]
-        (* Argument 1 is a matrix *)
+    [ ( "bernoulli_lpmf"
+      , [ [ (0, UnsizedType.DataOnly, UnsizedType.UArray UnsizedType.UInt)
+          ; (1, UnsizedType.DataOnly, UnsizedType.UReal) ] ] )
+    ; ( "bernoulli_logit_glm_lpmf"
+      , [ (* Array of conditions under which we do not want to move to OpenCL *)
+          [(1, UnsizedType.DataOnly, UnsizedType.URowVector)]
+        (* Argument 1 (0-based indexing) is a row vector *)
          ] )
-    ; ("bernoulli_logit_glm_lpmf", [[(1, UnsizedType.UMatrix)]])
-    ; ("categorical_logit_glm_lpmf", [[(1, UnsizedType.UMatrix)]])
-    ; ("neg_binomial_2_log_glm_lpmf", [[(1, UnsizedType.UMatrix)]])
-    ; ("ordered_logistic_glm_lpmf", [[(1, UnsizedType.UMatrix)]])
-    ; ("poisson_log_glm_lpmf", [[(1, UnsizedType.UMatrix)]]) ]
+    ; ( "categorical_logit_glm_lpmf"
+      , [[(1, UnsizedType.DataOnly, UnsizedType.URowVector)]] )
+    ; ( "exponential_lpdf"
+      , [ [ (0, UnsizedType.AutoDiffable, UnsizedType.UVector)
```

Comment:
(0, UnsizedType.AutoDiffable, UnsizedType.UVector) means that we will not move to OpenCL if the first argument is a parameter but will if its data.